### PR TITLE
add new pkg: sftp-go

### PR DESCRIPTION
**Pull request by vanlongme**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)
- [ ] If non-streamed package names no longer built, open PR to withdraw them (see [WITHDRAWING PACKAGES](https://github.com/wolfi-dev/os/blob/main/WITHDRAWING_PACKAGES.md))

#### For package updates (renames) in the base images

When updating packages part of base images (i.e. cgr.dev/chainguard/wolfi-base or ghcr.io/wolfi-dev/sdk)
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk images successfully build
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk contain no obsolete (no longer built) packages
- [ ] Upon launch, does `apk upgrade --latest` successfully upgrades packages or performs no actions

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: sftpgo.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package:
+  name: sftpgo
+  version: 2.6.2
+  epoch: 0
+  description: Full-featured and highly configurable SFTP, HTTP/S, FTP/S and WebDAV server - S3, Google Cloud Storage, Azure Blob
+  copyright:
+  - license: AGPL-3.0-only
+  dependencies:
+    provides:
```
Comment (tuananh) [inline sftpgo.yaml:9]: we dont need provide here unless we do version stream, if i understand correctly.

File: sftpgo.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    mkdir -p /usr/local/bin && ./docker/scripts/download-plugins.sh
+
+- runs: |
+    mkdir -p ${{targets.destdir}}/etc/sftpgo ${{targets.destdir}}/var/lib/sftpgo ${{targets.destdir}}/usr/share/sftpgo ${{targets.destdir}}/srv/sftpgo/data ${{targets.destdir}}/srv/sftpgo/backups ${{targets.destdir}}/usr/local/bin
+    cp sftpgo.json ${{targets.destdir}}/etc/sftpgo/sftpgo.json
+    cp -r templates ${{targets.destdir}}/usr/share/sftpgo/templates
+    cp -r static ${{targets.destdir}}/usr/share/sftpgo/static
+    cp -r openapi ${{targets.destdir}}/usr/share/sftpgo/openapi
+    cp sftpgo ${{targets.destdir}}/usr/local/bin/
+    cp /usr/local/bin/sftpgo-plugin-* ${{targets.destdir}}/usr/local/bin/
```
Comment (tuananh) [inline sftpgo.yaml:53]: move stuff under `usr/local` to `sftpgo-compat` package maybe?

```
2024/08/08 00:42:40 WARN linter "usrlocal" failed on package "sftpgo": /usr/local path found in non-compat package: usr/local/bin/sftpgo; suggest: This package should be a -compat package
```

**Comment by tuananh** [comment]
not sure if this an issue or not but the binary we build here show this in `--version` command => dirty

```
SFTPGo 2.6.2-636a1c2c-dirty-2024-08-08T00:37:07Z +metrics +azblob +gcs +s3 +bolt +mysql +pgsql +sqlite -unixcrypt +portable
```

whereas the official build show 

```
SFTPGo 2.6.2-636a1c2c-2024-06-21T17:30:20Z +metrics +azblob +gcs +s3 +bolt +mysql +pgsql +sqlite +unixcrypt +portable
```

**Comment by tuananh** [comment]
@vanlongme left a couple comments. other than that, lgtm.

File: sftpgo.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    sed -i 's|"users_base_dir": "",|"users_base_dir": "/srv/sftpgo/data",|' sftpgo.json && \
+    sed -i 's|"backups"|"/srv/sftpgo/backups"|' sftpgo.json && \
+    sed -i 's|"sqlite"|"bolt"|' sftpgo.json
+
+- runs: |
+    export GOFLAGS="-mod=readonly"
+    go mod download
+    set -xe && \
+    export COMMIT_SHA=${COMMIT_SHA:-$(git describe --always --abbrev=8 --dirty)} && \
+    go build -trimpath -ldflags "-s -w -X github.com/drakkan/sftpgo/v2/internal/version.commit=${COMMIT_SHA} -X github.com/drakkan/sftpgo/v2/internal/version.date=`date -u +%FT%TZ`" -v -o sftpgo
```
Comment (tuananh) [inline sftpgo.yaml:42]: can we use `go/build` instead?

File: sftpgo.yaml (hunk 2 above)
Comment (vanlongme) [reply sftpgo.yaml:53]: I already tried to switch to /usr/bin to fix that linter, still not work.

File: sftpgo.yaml (hunk 1 above)
Comment (vanlongme) [reply sftpgo.yaml:9]: As enabled update, I'd like to provide version stream for next version too

File: sftpgo.yaml (hunk 2 above)
Comment (tuananh) [reply sftpgo.yaml:53]: Move them to subpkg sftpgo-compat

File: sftpgo.yaml (hunk 1 above)
Comment (tuananh) [reply sftpgo.yaml:9]: version stream as is if you want to provide multiple versions of sftpgo in wolfi repo.

**Comment by vanlongme** [comment]
> @vanlongme left a couple comments. other than that, lgtm.

thanks, I already fixed and add new separated pkgs for official plugins.

**Comment by tuananh** [comment]
@ajayk can you help talking a look at this?

**Comment by tuananh** [comment]
@xnox can you help taking a look at this?

File: sftpgo-plugin-auth.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  - uses: go/bump
+    with:
+      deps: google.golang.org/grpc@v1.64.1
+
+  - uses: go/build
+    with:
+      packages: .
+      output: sftpgo-plugin-auth
+      ldflags: -s -w -X github.com/sftpgo/sftpgo-plugin-auth/cmd.commitHash=$(git describe --always) -X github.com/sftpgo/sftpgo-plugin-auth/cmd.buildDate=$(date -u +%FT%TZ)
```
Comment (xnox) [inline sftpgo-plugin-auth.yaml:34]: to support govulnchecker in full we typically build without "-s". Also the go/build pipeline automatically specifies "-w" already.

Thus "-s -w" should be dropped from this and all other files.

I guess I really must make that a failure in the go/build pipeline.

File: sftpgo-plugin-auth.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+environment:
+  contents:
+    packages:
+      - busybox
+      - ca-certificates-bundle
+      - curl
+      - git
+      - go
+      - wolfi-baselayout
```
Comment (xnox) [inline sftpgo-plugin-auth.yaml:17]: As pipelines are used for everything, I think the whole environment block can be dropped. I think this applies to all the files proposed here.

**Review by xnox — DISMISSED** [review]
Note -dirty is the result of melange git checkout pipeline adding .gitconfig files, and also as a result of go/bumps.

Thus it is correct to state that the version is dirty, as indeed some things got modified.

File: sftpgo-plugin-auth.yaml (hunk 4 above)
Comment (vanlongme) [reply sftpgo-plugin-auth.yaml]: thanks !

File: sftpgo-plugin-auth.yaml (hunk 5 above)
Comment (vanlongme) [reply sftpgo-plugin-auth.yaml]: I already fixed them, please help to review @xnox

**Comment by tuananh** [comment]
@xnox can you take another look? thanks

File: sftpgo.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      repository: https://github.com/drakkan/sftpgo
+      tag: v${{package.version}}
+      expected-commit: 636a1c2c385a16966842da4acad5acd36163ffb4
+
+  - runs: |
+      # Modify the default configuration file
+      sed -i 's|"users_base_dir": "",|"users_base_dir": "/srv/sftpgo/data",|' sftpgo.json && \
+      sed -i 's|"backups"|"/srv/sftpgo/backups"|' sftpgo.json && \
+      sed -i 's|"sqlite"|"bolt"|' sftpgo.json
+
```
Comment (xnox) [inline sftpgo.yaml:21]: Note to self compare what official dockerhub sftgo image ships, such that this build remains a drop-in substitute.

Comment (tuananh) [reply sftpgo.yaml:21]: yeah, we tried to make it drop-in replacement if possible.

**Comment by tuananh** [comment]
@xnox friendly ping. is there anything else we need to do here?

**Comment by tuananh** [comment]
@ajayk friendly ping. can you take a look?

**Review by ajayk — APPROVED** [review]

**Comment by ajayk** [comment]
> @ajayk friendly ping. can you take a look?

Can you please add tests as a follow up or

**Pull request merged**